internal/rpcproxy/rpcproxy.go

```diff
@@ -307,11 +308,18 @@ type modelProxy struct {
 	deviceOAuthResponse *oauth2.DeviceAuthResponse
 }
 
-func (p *modelProxy) sendError(socket *writeLockConn, req *message, err error) {
+func (p *modelProxy) sendError(ctx context.Context, socket *writeLockConn, req *message, err error) {
 	if req == nil {
 		// If there was no message to error on, just return.
 		return
 	}
+	if errors.ErrorCode(err) == errors.CodeUnauthorized {
+		logger.LogUnauthorizedAccess(
+			ctx,
+			p.tokenGen.GetUser().String(),
+			fmt.Sprintf("unauthorized access to model proxy for model %s", p.modelUUID),
```

Comment:
Slight change in wording and I'd log the error alongside the message, wdyt?
```suggestion
			fmt.Sprintf("unauthorized access in model proxy for model %s:", p.modelUUID),
```

Reply:
i wouldn't put the error along with the description.
We could make it as an `Extra` params if needed